http/ruby/nethttp/microsoft_kiota_nethttp/lib/microsoft_kiota_nethttp/http_core.rb

```diff
@@ -0,0 +1,85 @@
+require 'microsoft_kiota_abstractions'
+require 'net/https'
+require 'net/http'
+require 'concurrent'
+
+module MicrosoftKiotaNethttp
+  class HttpCore
+    include MicrosoftKiotaAbstractions::HttpCore
+    include Concurrent::Async
+
+    attr_accessor :authorization_header_key, :content_type_header_key, :parse_node_factory, :serialization_writer_factory, :client
+
+    def initialize(authentication_provider, parse_node_factory, serialization_writer_factory, client = Net::HTTP)
+      if !authentication_provider
+        raise StandardError , 'authentication provider cannot be null'
+      end
+      @authentication_provider = authentication_provider
+      @authorization_header_key = 'Authorization'
+      @content_type_header_key = 'Content-Type'
+      @parse_node_factory = parse_node_factory
+      @serialization_writer_factory = serialization_writer_factory 
```

Comment:
add a comment to get the static instance if factories are nil when implementing #478